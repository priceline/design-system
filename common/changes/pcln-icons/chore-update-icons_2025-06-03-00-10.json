{
  "changes": [
    {
      "packageName": "pcln-icons",
      "comment": "Update to styled-system v5 on pcln-icons, move to dev dependency and peer dependency. Use Node v20 for the build process.",
      "type": "minor"
    }
  ],
  "packageName": "pcln-icons"
}