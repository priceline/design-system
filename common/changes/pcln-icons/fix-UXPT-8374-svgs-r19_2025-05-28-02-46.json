{
  "changes": [
    {
      "packageName": "pcln-icons",
      "comment": "replace defaultProps usage in svgr template file",
      "type": "patch"
    }
  ],
  "packageName": "pcln-icons"
}