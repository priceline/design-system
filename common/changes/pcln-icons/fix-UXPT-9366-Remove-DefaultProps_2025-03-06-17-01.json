{
  "changes": [
    {
      "packageName": "pcln-icons",
      "comment": "remove defaultProps",
      "type": "minor"
    }
  ],
  "packageName": "pcln-icons"
}