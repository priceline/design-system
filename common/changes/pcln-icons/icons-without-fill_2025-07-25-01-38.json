{
  "changes": [
    {
      "packageName": "pcln-icons",
      "comment": "UXPT-10383: rounded Material SVGs from filled to unfilled",
      "type": "major"
    }
  ],
  "packageName": "pcln-icons"
}