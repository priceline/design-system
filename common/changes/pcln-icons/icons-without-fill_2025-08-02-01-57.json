{
  "changes": [
    {
      "comment": "allow styled-components 6 as a peer",
      "type": "patch",
      "packageName": "pcln-icons"
    }
  ],
  "packageName": "pcln-icons",
  "email": "steven.dalonzo@priceline.com"
}