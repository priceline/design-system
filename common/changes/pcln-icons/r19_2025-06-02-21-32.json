{
  "changes": [
    {
      "comment": "React 19 ecosystem upgrade",
      "type": "major",
      "packageName": "pcln-icons"
    }
  ],
  "packageName": "pcln-icons",
  "email": "steven.dalonzo@priceline.com"
}