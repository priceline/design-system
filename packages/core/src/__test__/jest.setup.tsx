import '@testing-library/jest-dom'
import 'jest-styled-components'

global.requestAnimationFrame =
  global.requestAnimationFrame ||
  function _raf(cb) {
    return setTimeout(cb, 0)
  }
