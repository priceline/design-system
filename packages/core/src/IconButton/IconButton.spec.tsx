import { Key as KeyIcon } from 'pcln-icons'
import { fireEvent, render } from '../__test__/testing-library'
import { IconButton } from './IconButton'

describe('IconButton', () => {
  test('executes onClick prop on click', () => {
    const handleClick = jest.fn()
    const { getByRole } = render(<IconButton icon={<KeyIcon />} onClick={handleClick} />)

    fireEvent.click(getByRole('button'))
    expect(handleClick).toHaveBeenCalled()
  })

  test('renders without background color in disabled state', () => {
    const { container } = render(<IconButton icon={<KeyIcon />} disabled />)

    const button = container.querySelector('button:disabled')
    expect(button).toHaveStyleRule('background-color', 'transparent')
  })

  test('renders without props', () => {
    const { asFragment } = render(<IconButton icon={<KeyIcon />} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
