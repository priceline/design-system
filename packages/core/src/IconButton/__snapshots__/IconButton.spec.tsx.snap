// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton renders without props 1`] = `
.c4 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  line-height: 1;
  width: 24px;
}

.c5 {
  outline: none;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 {
  -webkit-font-smoothing: antialiased;
  display: inline-block;
  vertical-align: middle;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: inherit;
  font-weight: 700;
  line-height: 1.5;
  cursor: pointer;
  border-radius: 9999px;
  border-width: 0;
  border-style: solid;
  border-radius: 9999px;
  font-size: 14px;
  padding: 9.5px 18px;
  background-color: #0068ef;
  color: #fff;
}

.c0:disabled {
  cursor: not-allowed;
  color: #879db3;
  background-color: #edf0f3;
}

.c0:hover {
  background-color: #049;
  color: #fff;
}

.c0:focus-visible {
  outline: 0px solid #049;
  box-shadow: 0 0 0 2px #049;
}

.c1 {
  padding: 0;
  background-color: transparent;
  color: inherit;
}

.c1:hover {
  background-color: transparent;
  color: inherit;
}

.c1 > div {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1:disabled {
  background-color: transparent;
}

<button
  className="c0 c1"
  color="primary"
  size="medium"
>
  <div
    className="c2"
  >
    <div
      className="c3"
    >
      <svg
        aria-hidden={true}
        className="c4 c5"
        fill="currentcolor"
        focusable={false}
        height={24}
        role="img"
        tabIndex={-1}
        viewBox="0 -960 960 960"
        width={24}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M280-400q-33 0-56.5-23.5T200-480q0-33 23.5-56.5T280-560q33 0 56.5 23.5T360-480q0 33-23.5 56.5T280-400zm0 160q-100 0-170-70T40-480q0-100 70-170t170-70q67 0 121.5 33t86.5 87h335q8 0 15.5 3t13.5 9l80 80q6 6 8.5 13t2.5 15q0 8-2.5 15t-8.5 13L805-325q-5 5-12 8t-14 4q-7 1-14-1t-13-7l-52-39-57 43q-5 4-11 6t-12 2q-6 0-12.5-2t-11.5-6l-61-43h-47q-32 54-86.5 87T280-240zm0-80q56 0 98.5-34t56.5-86h125l58 41v.5-.5l82-61 71 55 75-75h-.5.5l-40-40v-.5.5H435q-14-52-56.5-86T280-640q-66 0-113 47t-47 113q0 66 47 113t113 47z"
        />
      </svg>
    </div>
  </div>
</button>
`;
